gcloud-java-core/src/main/java/com/google/gcloud/BatchResult.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud;
+
+import static com.google.common.base.Preconditions.checkState;
+
+/**
+ * This class holds a single result of a batch call. {@code T} is the type of the result and
+ * {@code E} is the type of the service-dependent exception thrown when processing error occurs.
+ */
+public abstract class BatchResult<T, E extends BaseServiceException> {
+
+  private T result;
+  private boolean submitted = false;
+  private E error;
+
+  /**
+   * Returns {@code true} if the batch has been submitted and the result is available; {@code false}
```

Comment:
Nit: this is just my opinion, but the javadoc and method naming seem a little at odds; the javadoc says "true if submitted AND result available" while the method name is just "submitted".  Do you think it'd make sense to call this method something like "completed"?